fix(nuxt-module): Fixes nuxt pages

angelhdzdev opened the pull request:
- Fixes nuxt pages. 
- Added `consola : ^3.4.0` to show information about Vue Vine & SFC pages added.

ShenQingchuan commented:
@angelhdzdev 

Thanks for making this amazing contribution!

I've noticed that you're willing to add an option `pages: true (or false)` to toggle this Nuxt module for processing page features, but I don't really understand why you want to put those `.vine.ts` pages under path `/vine:excluded`, what does that mean?

angelhdzdev commented:
> @angelhdzdev
> 
> Thanks for making this amazing contribution! I'd like to merge this but I'm also aware that you may have some further changes ...
> 
> I've noticed that you're willing to add an option `pages: true (or false)` to toggle this Nuxt module for processing page features, but I don't really understand why you want to put those `.vine.ts` pages under path `/vine:excluded`, what does that mean?

Thank you a lot for the warm welcome! 

About hiding the routes behind `/vine:excluded`, yeah, I'm sorry for adding that confusion. 
This was something I desperately did at the last moment, because splicing the `.vine` pages from the array didn't work at first.
 
 I realized that because I deleted all Vine pages when `pages: false`, including `pages/index.vine.ts`, got a `404` error because the `pages` directory was left empty, in other words there were no routes for the Vue Router. So what I need to do is, add a warning when `pages: false` and the `pages` array ends up being empty of  pages, to remind the user to delete the `pages` directory if it's empty to avoid the 404 error. Something like "Warning: Vine Pages where disabled and no Nuxt pages where found.  Remove `pages` directory or set `pages: false` in Nuxt Config to disable routing".
 
### Motivation of `pages: boolean`
The motivation for adding `pages: boolean` module option is to enable or disable the inclusion of Nuxt pages that are Vine because of the philosophy of Vine of being "another style to write Vue" and "not to impose it", so the user can have `.vue` pages along side of `.vine.ts` pages, or totally disable Vine Nuxt pages and only have Vine components.

If we add this feature, then I also added the `useVueVineNuxt` (or just `useVueVine` if there are not plans to add an utility like this in a non Nuxt project) utility, that exposes the nuxt module options, in this case `pages: boolean` for conditional rendering of Vine pages.

This option is the equivalent of `pages: boolean` of Nuxt Config to enable/disable Nuxt pages, but for Vine pages.

ShenQingchuan commented:
Thanks for your detailed description.

So you mean:

in Vine Nuxt module config
- pages: true = if there's `pages/index.vine.ts` as Nuxt pages "/" route
- pages: false = if there's `pages/index.vine.ts`, set it to route "/vine:excluded/" to hide it, and print a warning.

Do I misunderstand anything?
I think this is a very good suggestion!

ShenQingchuan commented:
@angelhdzdev 

Great, and would you mind open another PR for these new changes?
We really want to record every new feature in separated threads.
